lib/evss/auth_headers.rb

```diff
@@ -49,8 +49,11 @@ def eauth_json
     end
 
     def iso8601_birth_date
-      return nil unless @user&.va_profile&.birth_date
-      DateTime.parse(@user.va_profile.birth_date).iso8601
+      if @user&.va_profile&.birth_date
+        DateTime.parse(@user.va_profile.birth_date).iso8601
+      elsif @user.birth_date
```

Comment:
Under what circumstances is a `@user` going to have a birth_date attribute directly vs. on the va_profile? And without the safe deference operator, mightn't this raise an exception instead of returning nil?

Reply:
It _seems_ like the only case where this happens is within this API. I can certainly add the `&` and leave an else case for returning nil since that was the default before when a birth date wasn't defined.

Reply:
Can we simulate the user.va_profile hierarchy via a nested OpenStruct when we create the user object, so that we don't have to have conditional logic in this base library? I agree the convention of user.va_profile is not a great internal interface but given it's there, I'd like to remain consistent.

Reply:
yep we can absolutely do that.

Reply:
resolved here: https://github.com/department-of-veterans-affairs/vets-api/pull/2650#commits-pushed-a6f6cfa